java.hints/src/org/netbeans/modules/java/hints/suggestions/Lambda.java

```diff
@@ -630,4 +663,27 @@ protected void performRewrite(TransformationContext ctx) throws Exception {
             }
         }
     }
+    
+    private static final class AddVarLambdaParameterTypes extends JavaFix {
+
+        public AddVarLambdaParameterTypes(CompilationInfo info, TreePath tp) {
+            super(info, tp);
+        }
+
+        @Override
+        protected String getText() {
+            return Bundle.FIX_addVarLambdaParameters();
+        }
+
+        @Override
+        protected void performRewrite(JavaFix.TransformationContext ctx) throws Exception {
+            if (ctx.getPath().getLeaf().getKind() == Kind.LAMBDA_EXPRESSION) {
+                LambdaExpressionTree let = (LambdaExpressionTree) ctx.getPath().getLeaf();
+                TreeMaker make = ctx.getWorkingCopy().getTreeMaker();
+                let.getParameters().forEach((var) -> {
+                    ctx.getWorkingCopy().rewrite(var.getType(), make.Type("var")); // NOI18N
```

Comment:
while rewriting already var-ed parameter to `var` seems like no-op, it can still accidentally reformat or change whitespaces around which is not expected: the user sees partially var-ed lambda, will trigger the suggestion to "change the rest", and the already OK code will be "touched". Depends on exact contents of the file, formatting settings etc. Better to avoid unnecessary nop rewrites, if the check for `var` type is simple.

Reply:
The suggestion hint work for implicit or explicit parameter type. In these scenario need to convert all parameter types to 'var'.
In case of mixed parameter type, it has compilation error (compiler.err.invalid.lambda.parameter.declaration) and suggestion hint will not show and write operation will not perform.

Reply:
@sdedic I added the changes respective to your review comment. Please review the changes and provide your feedback